tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -71,36 +81,74 @@ object ScatterGather {
     /**
       * Produces a new [[Scatter]] from the existing scatter.  Partitions the the scattered objects of type Result
       * according to the given discriminator function.  The discriminator function maps an individual object of type
-      * Result to a key of type K.  The partitions are each a sequence of objects of type Result that share the same key
-      * of type K when the discriminator function is applied.
+      * Result to a key of type Key.  The partitions are each a sequence of objects of type Result that share the same key
+      * of type Key when the discriminator function is applied.
       *
       * The resulting Scatter[(Key, Seq[Result])] can be further mapped or gathered (or both!).
       */
     def groupBy[Key](f: Result => Key) : Scatter[(Key, Seq[Result])]
-  }
 
+    /**
+      * Converts this [[Scatter]] of scattered scatterers into a [[Scatter]] formed by the elements of the scatterers.
+      * In other words, this scatter must be consist of elements that themselves are scatters of the resulting element
+      * type.
+      *
+      * @param toScatter an implicit conversion which asserts that the element type this scatter is a [[Scatter]].
+      * @tparam NextResult the type of element of each scattered [[Scatter]]
+      */
+    def flatten[NextResult](implicit toScatter: Result => /*<:<!!!*/ Scatter[NextResult]): Scatter[NextResult]
```

Comment:
Is the `<:<!!!` in a comment meaningful here?

Reply:
I'm not entirely sure having `flatten` in the API is useful other than letting you provide a default implementation of `flatMap()`.  E.g. in your example you could just have easily redefined the following two lines:

```scala
val scatterByLaneDemux: Scatter[Scatter[Library]] = scatterByLane.map { lane: Int => demux(lane=lane) }
val gatherLibararies: Scatter[Library]  = scatterByLane.flatten 
```

as

```scala
val scatterLibrary: Scatter[Library] = scatterByLane.flatMap { lane: Int => demux(lane=lane) }
```

And the only way you could end up with a need to `flatten()` given the types involved is via `map`, so could we stick to just `flatMap`?